packages/tpetra/core/test/Map/Map_getMinGlobalIndex.cpp

```diff
@@ -0,0 +1,212 @@
+//
+// Under the terms of Contract DE-AC04-94AL85000 with Sandia Corporation,
+// the U.S. Government retains certain rights in this software.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are
+// met:
+//
+// 1. Redistributions of source code must retain the above copyright
+// notice, this list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright
+// notice, this list of conditions and the following disclaimer in the
+// documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the Corporation nor the names of the
+// contributors may be used to endorse or promote products derived from
+// this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY SANDIA CORPORATION "AS IS" AND ANY
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+// PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL SANDIA CORPORATION OR THE
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+// EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+// PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+// LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+// NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+// SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Questions? Contact Michael A. Heroux (maherou@sandia.gov)
+//
+// ************************************************************************
+// @HEADER
+*/
+
+// Some Macro Magic to ensure that if CUDA and KokkosCompat is enabled
+// only the .cu version of this file is actually compiled
+#include <Tpetra_ConfigDefs.hpp>
+#include <Tpetra_Map.hpp>
+#include <Tpetra_TestingUtilities.hpp>
+#include <type_traits> // std::is_same
+
+// FINISH: add testing of operator==, operator!=, operator=, copy construct
+// put these into test_same_as and test_is_compatible
+
+namespace {
+
+  using Teuchos::null;
+
+  using Tpetra::TestingUtilities::getNode;
+  using Tpetra::TestingUtilities::getDefaultComm;
+
+  using Tpetra::createUniformContigMapWithNode;
+
+  using Teuchos::Array;
+  using Teuchos::ArrayView;
+  using Teuchos::as;
+  using Teuchos::RCP;
+  using Teuchos::arcp;
+  using Teuchos::rcp;
+  using Teuchos::outArg;
+  using Teuchos::Tuple;
+  using Teuchos::tuple;
+  using Tpetra::Map;
+  using Tpetra::global_size_t;
+  using Tpetra::DefaultPlatform;
+  using std::sort;
+  using std::find;
+  using Teuchos::broadcast;
+  using Teuchos::OrdinalTraits;
+  using Teuchos::Comm;
+  using std::endl;
+
+  ////
+  TEUCHOS_UNIT_TEST_TEMPLATE_2_DECL( Map, getMinGlobalIndex_nonuniform, LO, GO )
+  {
+    typedef Map<LO,GO> Map;
```

Comment:
Please don't give a typedef the same name as a type; I'm not sure that's legit C++.

Reply:
Oh wait, MueLu normally does that.  I don't like it but if it's legit C++, I won't complain.

Reply:
Actually: If MueLu's "short names" normally does this, could this collide with MueLu's short names?  If so, please don't do it :-) .

Reply:
Yeah, I lazily copied that bit from an other unit-test in `Tpetra`. I removed that typedef all together since I only use it once in the test.